From 03580b92b279b0aa125df3d8c805f340597c9ee2 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:44 -0400
Subject: [tools] perf/lock: Limit bad rate precision

Message-id: <1408462094-14194-116-git-send-email-jolsa@redhat.com>
Patchwork-id: 88091
O-Subject: [PATCH RHEL7.1 BZ1131394 115/325] perf lock: Limit bad rate precision
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 60a25cbc4a167fc0129296c3c640d8506a57acc5
Author: Davidlohr Bueso <davidlohr@hp.com>
Date: Sun Sep 8 19:19:18 2013 -0700

description
===========
Two decimal precision should be enough for this.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-lock.c b/tools/perf/builtin-lock.c
index 780484f..972310c 100644
--- a/tools/perf/builtin-lock.c
+++ b/tools/perf/builtin-lock.c
@@ -686,7 +686,7 @@ static void print_bad_events(int bad, int total)
 
  pr_info("\n=== output for debug===\n\n");
  pr_info("bad: %d, total: %d\n", bad, total);
- pr_info("bad rate: %f %%\n", (double)bad / (double)total * 100);
+ pr_info("bad rate: %.2f %%\n", (double)bad / (double)total * 100);
  pr_info("histogram of events caused bad sequence\n");
  for (i = 0; i < BROKEN_MAX; i++)
   pr_info(" %10s: %d\n", name[i], bad_hist[i]);
-- 
1.7.1